_clients/dot-net-conventions.md

```diff
@@ -0,0 +1,90 @@
+---
+layout: default
+title: .NET client conventions
+nav_order: 20
+has_children: false
+parent: .NET clients
+---
+
+# .NET client conventions and best practices
+
+## Registering OpenSearch.Client as a singleton
+
+As a rule, you want to set up your OpenSearch.Client as a singleton. OpenSearch.Client manages connections to the server and the states of the nodes in a cluster. Additionally, each client uses a lot of configuration for its setup. Therefore, it is beneficial to create an OpenSearch.Client instance once and reuse it for all OpenSearch operations. The client is thread-safe, so the same instance can be shared by multiple threads.
+
+## Exceptions
+
+The following are the types of exceptions that may be thrown by .NET clients:
+
+- `OpenSearchClientException` is a known exception that occurs either in the request pipeline (for example, timeout reached) or in OpenSearch (for example, malformed query). If it is an OpenSearch exception, the `ServerError` response property contains the error that OpenSearch returns. 
+- `UnexpectedOpenSearchClientException` is an unknown exception (for example, an error during deserialization) and is a subclass of OpenSearchClientException.
+- System exceptions are thrown when the API is not used properly. 
+
+## Nodes
+
+To create a node, pass a `Uri` object into its constructor:
+
+```cs
+var uri = new Uri("http://example.org/opensearch");
+var node = new Node(uri);
+```
+
+When first created, a node is master eligible and its `HoldsData` property is set to true. 
+The `AbsolutePath` property of the node created above is `"/opensearch/"`: a trailing forward slash is appended so that the paths can be easily combined. If not specified, the default `Port` is 80.
+
+Nodes are considered equal if they have the same endpoint. Metadata is not taken into account when checking nodes for equality.
+{: .note}
+
+## Connection pools
+
+Connecttion pools are instances of `IConnectionPool` and are responsible for managing the nodes in the OpenSearch cluster. We recommend creating a [singleton client](#registering-the-osc-client-as-a-singleton) with a single `ConnectionSettings` object. The lifetime of both the client and its `ConnectionSettings` is the lifetime of the application.
+
+The following are connection pool types.
+
+- **SingleNodeConnectionPool**
+
+`SingleNodeConnectionPool` is the default connection pool that is used if no connection pool is passed to the `ConnectionSettings` constructor. Use a `SingleNodeConnectionPool` if you have only one node in the cluster or if your cluster has a load balancer as an entry point. `SingleNodeConnectionPool` does not support sniffing or pinging and does not mark nodes dead or alive. 
```

Comment:
```suggestion
`SingleNodeConnectionPool` is the default connection pool that is used if no connection pool is passed to the `ConnectionSettings` constructor. Use `SingleNodeConnectionPool` if you have only one node in the cluster or if your cluster has a load balancer as an entry point. `SingleNodeConnectionPool` does not support sniffing or pinging and does not mark nodes dead or alive. 
```

Reply:
@hdhalter - thank you for the feedback. I implemented your comments.